test/accessibility.js

```diff
@@ -0,0 +1,437 @@
+      var title = this.element.querySelector('h1');
+
+      title.textContent = 'really long title really long title really long title';
+      otherButton.textContent = 'another button';
+      this.element.appendChild(otherButton);
+    });
+  });
+
+  suite('Title centering:', function() {
+    teardown(function(done) {
+      accessibility.check(this.dom).then(done, done);
+    });
+
+    test('It centers the header between the surrounding buttons', function(done) {
+      this.dom.innerHTML = `<gaia-header action="menu">
+        <h1>Header title</h1>
+        <button>Foo</button>
+        <button>Bar</button>
+      </gaia-header>`;
+
+      var el = this.dom.firstElementChild;
+      var h1 = el.querySelector('h1');
+
+      afterNext(el, 'runFontFit').then(() => accessibility.check(this.dom)
+      ).then(() => {
+        this.dom.innerHTML = `<gaia-header action="menu">
+          <h1>Header title</h1>
+        </gaia-header>`;
+
+        el = this.dom.firstElementChild;
+        h1 = el.querySelector('h1');
+
+        return afterNext(el, 'runFontFit');
+      }).then(done, done);
+    });
+
+    test('It doesn\'t center if there is not enough space', function(done) {
+      this.dom.innerHTML = `<gaia-header action="menu">
+        <h1>This title is far far far far far far far far far far far far
+        far far too long to center</h1>
+        <button>Foo</button>
+        <button>Bar</button>
+      </gaia-header>`;
+
+      afterNext(this.dom.firstElementChild, 'runFontFit').then(done, done);
+    });
+
+    test('It doesn\'t have start/end padding if title is centerable', function(done) {
+      this.dom.innerHTML = `<gaia-header action="menu">
+        <h1>Title can be center</h1>
+        <button>Foo</button>
+      </gaia-header>`;
+
+      afterNext(this.dom.firstElementChild, 'runFontFit').then(done, done);
+    });
+
+    test('It should apply 10px paddings when overflowing and no buttons before/after',
+      function(done) {
+      this.dom.innerHTML = `<gaia-header>
+        <h1>This title is far far far far far far far far far far far
+        far far far far far far far too long to center</h1>
+      </gaia-header>`;
+
+      afterNext(this.dom.firstElementChild, 'runFontFit').then(done, done);
+    });
+  });
+
+  test('It works with multiple headers', function(done) {
+    this.dom.innerHTML = `<gaia-header action="back">
+      <h1 class="title-1">Title 1</h1>
+      <h1 class="title-2" hidden>Title 2</h1>
+      <button>Foo</button>
+      <button>Bar</button>
+    </gaia-header>`;
+
+    var el = this.dom.firstElementChild;
+    var title1 = el.querySelector('.title-1');
+    var title2 = el.querySelector('.title-2');
+
+    afterNext(el, 'runFontFit').then(() => accessibility.check(this.dom)
+    ).then(() => {
+      this.fontFit.reset();
+      title1.hidden = true;
+      title2.hidden = false;
+      return afterNext(el, 'runFontFit');
+    }).then(done, done);
+  });
+});
```

Comment:
It seems a bit overkill to copy every unit-test and assert that the header is accessible after each one. I'd prefer to just have a `suite('accessiblity')` inside `test.js` when we perform a few well targeted accessibility tests.

It would also be interesting to know what the accessibility tests were actually checking for. For example `test('the action button is visible to a screen reader'), if my patch breaks one of these accessibility tests I will have no idea what is broken, so won't know how to fix it.